apps/client/src/features/modals/sheets-modal/SheetsModal.tsx

```diff
@@ -0,0 +1,250 @@
+      return;
+    } else {
+      await uploadSheetClientFile(selectedFile);
+      _onChange();
+    }
+  };
+
+  const _onChange = async () => {
+    setSheetState(await getSheetstate());
+  };
+
+  if (isOpen) {
+    //TODO: how to get this on modal open
+    getSheetSettings().then((data) => {
+      if (sheetid.current?.value != data.id || worksheet.current?.value != data.worksheet) {
+        _onChange();
+      }
+      if (sheetid.current) {
+        sheetid.current.value = data.id;
+      }
+      if (worksheet.current) {
+        worksheet.current.value = data.worksheet;
+      }
+    });
+  }
+
+  const handelSave = () => {
+    postSheetSettings({ id: sheetid.current?.value ?? '', worksheet: worksheet.current?.value ?? '' }).then((data) => {
+      _onChange();
+      if (sheetid.current) {
+        sheetid.current.value = data.id;
+      }
+      if (worksheet.current) {
+        worksheet.current.value = data.worksheet;
+      }
+    });
+  };
+
+  const handleAuthenticate = () => {
+    getSheetsAuthUrl().then((data) => {
+      if (data != 'bad') {
+        window.open(data, '_blank', 'noreferrer');
+      }
+    });
+  };
+
+  const handlePullData = () => {
+    postPreviewSheet().then((data) => {
+      setProject(data.project);
+      setRundown(data.rundown);
+      setUserFields(data.userFields);
+    });
+  };
+
+  const handlePushData = () => {
+    postPushSheet().then((data) => {
+      console.log(data);
+    });
+  };
+
+  const handleFinalise = async () => {
+    // this step is currently only used for excel files, after preview
+    if (rundown && userFields && project) {
+      let doClose = false;
+      try {
+        await patchData({ rundown, userFields, project });
+        queryClient.setQueryData(RUNDOWN, rundown);
+        queryClient.setQueryData(USERFIELDS, userFields);
+        queryClient.setQueryData(PROJECT_DATA, project);
+        await queryClient.invalidateQueries({
+          queryKey: [...RUNDOWN, ...USERFIELDS, ...PROJECT_DATA],
+        });
+        doClose = true;
+      } catch (error) {
+        const message = maybeAxiosError(error);
+        console.log(message);
+        // setErrors(`Failed applying changes ${message}`);
+      } finally {
+        if (doClose) {
+          handleClose();
+        }
+      }
+    }
+  };
+
+  return (
+    <Modal
+      onClose={handleClose}
+      isOpen={isOpen}
+      closeOnOverlayClick={false}
+      motionPreset='slideInBottom'
+      size='xl'
+      scrollBehavior='inside'
+      preserveScrollBarGap
+      variant='ontime-upload'
+    >
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Sheets!</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          {rundown && (
+            <>
+              <PreviewExcel
+                rundown={rundown ?? []}
+                project={project ?? projectDataPlaceholder}
+                userFields={userFields ?? userFieldsPlaceholder}
+              />
```

Comment:
I wonder whether we should skip this for now

It is complicating your code. This is an experimental feature and the main focus should be in the feature core. Once that works well and we have a good idea of how to integrate, we can expand.

This means we can move faster and be more decisive, for all we know, we decide that this logic does not belong here or that the feature is not usable for ontime